### ISSUE TITLE
Stuck on loading

### ISSUE THREAD

**Issue by Thomiehawk** [theothernt/AerialViews#125]
**What were you trying to do...**

Start screensaver after idling when watching something on kodi. 

**And what actually happened (ie. describe the bug)...**

Aerial views app opens and it displays the text: "Loading..." indefinitely

**What kind of device do you have? (eg. Nvidia Shield, Sony TV, etc)...**

Fire cube 3rd gen

Seems like this only happens with 4K videos for some reason.
When playing a 1080p video, screensaver kicks in normally.

**Comment by theothernt** [comment]
How long has this been happening - has there been a recent update to the Fire TV Cube or Kodi?

**Comment by Thomiehawk** [comment]
It happened to me ever since I got the fire cuben which is like 2 months ago. When I got the firmware was PS7299/3052 and now I am on the latest, issues persisted on both. I have tried kodi nexus and omega and am currently on one of the latest nightly builds of omega. I am on the same kodi version on my shield pro and it is working there.

Just tested, SDR does seem to work

**Comment by theothernt** [comment]
Thanks for the info, and to test one more thing - does 1080p Dolby Vision videos work?

Also, when you pause Kodi - do you happen to be watching Dolby Vision content?

**Comment by Thomiehawk** [comment]
No 1080p does not work either, I was watching dolby vision content indeed.

**Comment by theothernt** [comment]
This may be a compatibility issue with how Kodi handles Dolby Vision on the Fire TV Cube. Over the past couple of years, there have been several issues reported involving Kodi - either crashing or just having issues with Aerial Views running properly - and I'm not sure there is anything I can do. 

My guess with this current issue is that Kodi is not 'giving back' the Dolby Vision decoder when it pauses video playback so Aerial Views simply can't play that format. So the fix would be to use the SDR versions of the video.

Also, if you wanted to confirm the issue is with Kodi - I'd try downloading [Just (Video) Player](https://play.google.com/store/apps/details?id=com.brouken.player&hl=en), which is free, then playing a Dolby Vision clip, pausing it then waiting for Aerial Views to start. If the issue still occurs then the problem lies with how the Fire TV is handling it's video codecs or playback.

**Comment by Thomiehawk** [comment]
I have tried with just player and a dolby vision file. It did work, however with same kodi on nvidia shield and same scenario, Aerial views does work.

Which makes me believe it is still a fire tv incompatibility issue.

**Comment by theothernt** [comment]
At least it narrows it down a bit. So maybe the Kodi devs can do something, but otherwise you'll have to use SDR videos on the Fire TV?

**Comment by Thomiehawk** [comment]
Yes I suppose that would work. I have tried setting a timer to just quit kodi and start screensaver couple minutes after but haven't had much luck yet.
